### PR TITLE
[UC13#158#159] Implemented Edit Mode Button for users to allow modifying their owned/wished existing physical cards

### LINKED ISSUE #158: Rendere editabile PhysicalCardWidget per modificare le proprietà della carta nel mazzo delle carte possedute [UC13]
_**Dod**_
- [x]  La pull request ha un titolo e descrizione adeguata
- [x]  La pull request è stata collegata con la issue del task
- [x]  I requisiti funzionali del task collegato sono soddisfatti
- [x]  Il codice si compila senza errori
- [x]  Gli Unit Tests sono stati scritti e passati 
- [x]  Gli Unit Tests esistenti sono stati passati
- [ ]  Il Code Review è stato fatto
- [ ]  La documentazione è stata aggiornata
- [ ]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### LINKED ISSUE #159: Rendere editabile PhysicalCardWidget solo nel mazzo delle carte possedute e delle desiderate non negli altri mazzi [UC13]
_**Dod**_
- [x]  La pull request ha un titolo e descrizione adeguata
- [x]  La pull request è stata collegata con la issue del task
- [x]  I requisiti funzionali del task collegato sono soddisfatti
- [x]  Il codice si compila senza errori
- [x]  Gli Unit Tests sono stati scritti e passati 
- [x]  Gli Unit Tests esistenti sono stati passati
- [ ]  Il Code Review è stato fatto
- [ ]  La documentazione è stata aggiornata
- [ ]  La pull request è stata accettata e fatto merge sul branch development.

 **Effort effettivo:**

### PR DESCRIPTION
**Pull request by alessioarcara**
This PR implements #158 and #159. Adding an Edit Button for modification of existing Physical Deck Cards. The button puts the physical cards in the Decks View into an editable state, allowing users to modify the description and status fields. This button is only visible on owned-wished default decks."

### CODE REVIEW

**Review by Matte0Sacc0 — APPROVED** [review]
Good job and nice to have a specific widget for the status so it can be displayed in the same way in the entire application

File: src/main/java/com/aadm/cardexchange/client/views/DecksViewImpl.java [hunk 1]
```diff
@@ -21,9 +21,10 @@
 import java.util.function.BiConsumer;
 import java.util.function.Consumer;
 
-public class DecksViewImpl extends Composite implements DecksView, ImperativeHandleDeck {
+public class DecksViewImpl extends Composite implements DecksView, ImperativeHandleDeck, ImperativeHandleDeckRemove,
+        ImperativeHandlePhysicalCardEdit {
     private static final DecksViewImpl.DecksViewImplUIBinder uiBinder = GWT.create(DecksViewImpl.DecksViewImplUIBinder.class);
-    private static final String DEFAULT_CUSTOM_DECK_TEXT = "Write here name for your custom deck";
+    private static final String DEFAULT_CUSTOM_DECK_TEXT = "Write here name for your new custom deck";
```
Comment (alessiacrimaldi) [inline src/main/java/com/aadm/cardexchange/client/views/DecksViewImpl.java:27]: I'd prefer to set here a different placeholder text, like I showed you in the last Decks mock-up. May you fix this in future?

image

**Review by alessiacrimaldi — APPROVED** [review]
It seems ok to me. Could you please look at the comments left? Thank you.

File: src/main/java/com/aadm/cardexchange/client/views/DecksViewImpl.java (hunk 1 above)
Comment (alessioarcara) [reply src/main/java/com/aadm/cardexchange/client/views/DecksViewImpl.java:27]: Sure!

**Pull request merged**